homeassistant/helpers/intent.py

```diff
@@ -252,13 +265,47 @@ def __init__(
         self.text_input = text_input
         self.context = context
         self.language = language
+        self.category = category
 
     @callback
     def create_response(self) -> IntentResponse:
         """Create a response."""
         return IntentResponse(self, language=self.language)
 
 
+class IntentResponseState(Enum):
+    """State of the intent response."""
+
+    SUCCESS = "success"
+    FAILURE = "failure"
+
+
+class IntentResponseType(Enum):
+    """Type of the intent response."""
+
+    ACTION_CONFIRMATION = "action_confirmation"
+    """speech is a summary of action taken"""
```

Comment:
I would rename this to `ACTION_DONE`. Confirmation sounds too much like a reprompt about "Are you sure you want to do X"

Reply:
Or `ACTION_RESULT`

Reply:
It's confusing that the documentation references `speech` while an intent response can contain multiple things.